runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4001,23 +4001,51 @@ class INTERPRETER_CLASS
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native <V> V uninitializedDefaultValue(Class<?> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeUninitializedDefaultValue(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		j9object_t result = NULL;
+
+		/* ToDo: update this function to have the same behavior as OpenJDK when cls is null or not a vlauetype (currently OpenJDK segfaults in both those scenarios) */
+		if (NULL != cls) {
+			J9Class *j9clazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, cls);
+			if (J9_IS_J9CLASS_VALUETYPE(j9clazz)) {
+				result = j9clazz->flattenedClassCache->defaultValue;
+			}
+		}
+
+		restoreInternalNativeStackFrame(REGISTER_ARGS);
+		returnObjectFromINL(REGISTER_ARGS, result, 2);
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native <V> long valueHeaderSize(Class<V> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeValueHeaderSize(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		I_64 result = (I_64)0;
```

Comment:
is the result always zero if the class is not a VT?

Reply:
When I tested OpenJDK they just segfaulted when the class wasn't a VT. So I am using 0 for now until OpenJDK updates their behaviour (This is what the ToDo comment on the next line is referring to)